packages/terra-notification-dialog/src/NotificationDialog.jsx

```diff
@@ -0,0 +1,85 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import AbstractModal from 'terra-abstract-modal';
+import Button from 'terra-button';
+import classNames from 'classnames/bind';
+import 'terra-base/lib/baseStyles';
+import { elementType } from 'airbnb-prop-types';
+import styles from './NotificationDialog.scss';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+   * Title of the notification-dialog.
+   */
+  title: PropTypes.string,
+  /**
+   * Message of the notification-dialog.
+   */
+  message: PropTypes.string,
+  /**
+   * Array of buttons. Restricted to terra-button.
+   */
+  actions: PropTypes.arrayOf(elementType(Button)),
+  /**
+   * Toggle to show notification-dialog or not.
+   */
+  showNotificationDialog: PropTypes.bool.isRequired,
+  /**
+   * Callback function indicating a close condition was met, should be combined with showNotificationDialog for state management.
+   */
+  onRequestClose: PropTypes.func.isRequired,
+};
+
+const defaultProps = {
+  title: null,
+  message: null,
+  actions: [],
+};
+
+const actionSection = (actions) => {
+  if (!actions || !actions.length) {
+    return null;
+  }
+  return <div className={cx('actions')}>{actions}</div>;
+};
+
+class NotificationDialog extends React.Component {
+  render() {
+    if (!this.props.showNotificationDialog) {
+      return null;
+    }
+
+    const {
+      title,
+      message,
+      actions,
+      showNotificationDialog,
+      onRequestClose,
+      ...customProps
+    } = this.props;
+
+    return (
+      <div {...customProps} className={cx('notification-dialog')}>
```

Comment:
I almost think this outer div should be removed (if it can be) and the custom props spread onto the div with the class `notification-dialog-body`.

Right now this renders an empty div as a sibling to the triggering target, while all of the content is rendered into a portal somewhere else on the dom.

![image](https://user-images.githubusercontent.com/6720991/39365838-ece079c4-49f7-11e8-9be3-241e4519f630.png)